torch/csrc/jit/codegen/cuda/lower2device.cpp

```diff
@@ -174,20 +174,12 @@ Statement* GPULower::mutate(ReductionOp* rop) {
       ir_utils::isTVOp(rop),
       "Cannot have a reduction operation on something other than a tensor view.");
   auto loops = scope_utils::getLoops(active_scope);
-  TORCH_INTERNAL_ASSERT(
-      std::none_of(
-          loops.begin(),
-          loops.end(),
-          [](ForLoop* fl) {
-            return fl->iter_domain()->isBlockDim() &&
-                fl->iter_domain()->isReduction();
-          }),
-      "Reduction on block axes not yet supported.");
-
-  bool is_thread_reduce =
-      std::any_of(loops.begin(), loops.end(), [](ForLoop* fl) {
-        return fl->iter_domain()->isThreadDim() &&
-            fl->iter_domain()->isReduction();
+
+  bool is_private_reduce =
+      std::none_of(loops.begin(), loops.end(), [](ForLoop* fl) {
+        return fl->iter_domain()->isReduction() &&
+            (fl->iter_domain()->isThreadDim() ||
+             fl->iter_domain()->isBlockDim());
```

Comment:
Nit: you could use `fl->iter_domain()->isThread()` which is simply `isThreadDim() || isBlockDim()` on the IterDomain.